### ISSUE TITLE
Unable to find org.scoverage.ScoverageReport

### ISSUE THREAD

**Issue by fsarradin** [scoverage/gradle-scoverage#13]
Hi,

With the configuration displayed in the README.md file at the root of the project, gradle says that it is unable to find or load the main class `org.scoverage.ScoverageReport`. By launching gradle with `--debug` option, it says:

```
Starting process 'command '/Library/Java/JavaVirtualMachines/jdk1.7.0_45.jdk/Contents/Home/bin/java''.
Working directory: /Users/fsarradin/ks-framework/kff-coresetup
Command: /Library/Java/JavaVirtualMachines/jdk1.7.0_45.jdk/Contents/Home/bin/java -Dfile.encoding=UTF-8 -cp /Users/fsarradin/.gradle/caches/modules-2/files-2.1/org.scoverage/scalac-scoverage-plugin_2.11/0.99.5/ee02c604b73c8b6cca7c345a2fd419531da47d08/scalac-scoverage-plugin_2.11-0.99.5.jar:/Users/fsarradin/.gradle/caches/modules-2/files-2.1/org.scala-lang/scala-library/2.11.0/aa8f7553253318c468f90ce58a85f94bd1a224eb/scala-library-2.11.0.jar:/Users/fsarradin/.gradle/caches/modules-2/files-2.1/commons-io/commons-io/2.4/b1b6ea3b7e4aa4f492509a4952029cd8e48019ad/commons-io-2.4.jar:/Users/fsarradin/.gradle/caches/modules-2/files-2.1/org.scala-lang.modules/scala-xml_2.11/1.0.1/21dbac0088b91b8ffb7ac385301f4340f8ebe71f/scala-xml_2.11-1.0.1.jar org.scoverage.ScoverageReport /Users/fsarradin/ks-framework/kff-coresetup/src/main/scala /Users/fsarradin/ks-framework/kff-coresetup/build/scoverage /Users/fsarradin/ks-framework/kff-coresetup/build/reports/scoverage
```

So gradle-scoverage does not appear in the classpath. But I don't how to do it? Any idea?

I'm using gradle 2.0.

regards

**Comment by avdv** [comment]
I just tested with gradle-scoverage 1.0.6 and scoverage 1.0.1 and 1.0.2. But the error is still there. I guess I misinterpreted what 6686446 was meant to do.

Note, that I also tried Release 1.0 as you suggested, but it seems that was build with java 8 and I only have jdk 7.

Looking at https://github.com/scoverage/gradle-scoverage/blob/66864461947e207374fb15636e9f3c41ab07f187/src/main/groovy/org/scoverage/ScoverageExtension.groovy#L116 it seems that it should work, but printing the value of project.buildscript.configurations.classpath shows up empty. And consequently, the gradle-scoverage jar is missing from reportScoverage.classpath.

The discussion here (http://forums.gradle.org/gradle/topics/share_classpath_between_buildscript_and_project_configuration) seems to suggest that accessing the classpath configuration of the buildscript is not possible.

**Comment by maiflai** [comment]
Hi - thanks for reporting back - we've since released 1.0.6 which supports jdk 7, but I agree that I can reproduce an error here.

It appears to work correctly with a simple project, but please could I check that you have a multi-module project? It would also be helpful if you could share your build configuration.

Thanks again,
Stu.

**Comment by avdv** [comment]
Yes, I do have a multi-module project, but my parent project does have code. And btw, I'm using Gradle 2.2.1.

In my rootProject I'm using

```
apply from: "gradle/scoverage.gradle"
````

where the scoverage.gradle file contains the code from the "Gettting started" section of README.md.

When I include the code directly in my build.gradle file, it works fine. Is this a limitation of Gradle's apply mechanism, or a feature?

**Comment by maiflai** [comment]
That's interesting - are you applying this to `allprojects`, `subprojects`, or configuring individual projects?

**Comment by avdv** [comment]
I'm only configuring scoverage for the root project, the single subproject is not configured for scoverage yet.

When I try to do it for all projects:

```groovy
// root project

allprojects {
  buildscript {
    repositories {
      mavenCentral()
      maven { url "https://oss.sonatype.org/content/groups/public" }
    }
    dependencies {
      classpath 'org.scoverage:gradle-scoverage:1.0.6'
    }
  }
  apply plugin: org.scoverage.ScoveragePlugin
}
```
I get this error:

```
A problem occurred evaluating root project 'root'.
> Cannot change configuration ':classpath' after it has been resolved.
```
Which is OK, since the buildscripts have already been processed when the allprojects call is seen in the second phase of Gradle's build evaluation.

Then, I tried:

```groovy
// root project
buildscript {
  repositories {
    mavenCentral()
    maven { url "https://oss.sonatype.org/content/groups/public" }
  }
  dependencies {
    classpath 'org.scoverage:gradle-scoverage:1.0.6'
  }
}

allprojects {
  apply plugin: org.scoverage.ScoveragePlugin

  dependencies {
    scoverage 'org.scoverage:scalac-scoverage-plugin_2.11:1.0.2', 'org.scoverage:scalac-scoverage-runtime_2.11:1.0.2'
  }
}
```

Which yields this error in the subproject again:

```
Error: Could not find or load main class org.scoverage.ScoverageReport
```

**Comment by avdv** [comment]
Just looked through the gradle forums and I found this discussion: http://forums.gradle.org/gradle/topics/understanding_the_additivity_of_the_buildscript_dependencies

> The buildscript block has to be in the "main" build script, not the applied one. It's a known limitation.

So, the second script I posted above is the way to go, too bad it doesn't work for the subprojects.

**Comment by maiflai** [comment]
I've published version 1.0.8, which I hope addresses this issue.

Please could you upgrade and remove the workaround (if you used one) to test it?

**Comment by avdv** [comment]
Yes, works for me without having any workaround code in my build. Thanks!
